span.go

```diff
@@ -58,3 +62,30 @@ type Span interface {
 	// TraceContext.
 	AddToGoContext(goCtx context.Context) (Span, context.Context)
 }
+
+// See Span.Log(). Every LogData instance should specify at least one of Event
+// and/or Payload.
+type LogData struct {
```

Comment:
I really think we should start a page on the `.io` site where we define the Scope of the version X of the API by listing the Use Cases and the "canonical" way of implementing them via the API. Then all API changes can be validated on whether they satisfy the established use cases.

For example, the implication of the previous `Error()` log function was that the span is explicitly marked as "failed", i.e. an indication to the UI to display the span or the log entry differently, and possibly for sampler to capture the trace.